### PR TITLE
critical feature additions and testing improvements

### PR DESCRIPTION
**Pull request by appellation**
**Please describe the changes this PR makes and why it should be merged**:
- supports more types!
- tests those types!

**Semantic versioning classification:**  
- This PR changes the library's code in some way
  - [ ] SEMVER patch (bug fix)
  - [x] SEMVER minor (methods or parameters added)
  - [ ] SEMVER major (methods or parameters removed or renamed)
- [ ] This PR **only** includes non-code changes, like changes to the README.

### CODE REVIEW

**Comment by dragonfire535** [comment]
The purpose of `is-tuesday` is to be a validator for if the current day of the week according to system time is Tuesday. This is a very complex and sophisticated process with lots of chance of failure, hence the heavy testing system. Adding more to that is going to be too complex for one repository to handle.

Checking for the string and object `valueOf` tuesday would be better placed in it's own module, according to the small module philosophy, as every single use-case should be it's own module, until you get to the big project. `is-tuesday-string`, `is-tuesday-object`, `is-tuesday-date`, and `is-tuesday-now` should all be created (with their own tests and typings) and then come together in a big `is-tuesday` wrapper. This aligns with current practice on npm and overall small module philosophy.

tl;dr, I think this should be forked off into it's own repository and then split into separate modules, which are then exported by this one.

**Comment by appellation** [comment]
The purpose of a loosely typed language like Javascript is that you can perform checks on multiple types of data in the same method.  This allows for greatly simplified code since you don't have to perform the same checks multiple times across modules.  Further, while the concept of a monorepo has been lately derided as outmoded and inefficient, I think that in this case we will see significant benefit from having all our code in even the same function.  While I do realize that scrolling through tens of lines just to find a particular bug in the code is annoying, imagine having to sort through multiple files or even projects just to get to a specific chunk of code.

Imagine downloading a library named `is-tuesday` and then calling it with a timestamp.  You would not expect this library to still use the current date; you clearly specified a timestamp and the library should be able to figure out your intent based on the type of parameter you've provided.  This PR adds this braindead functionality so that even the noobiest of coders can operate this library without random `TypeError`s or `cannot read property x of undefined`.  We cater to the typical Javascript coder, and we must be certain to understand their needs.

Further, while I can see future issues with high-volume users needing to microservice this into separate distinct servers to handle the different data types, personally I think they should be perfectly OK just downloading more RAM.  This is the accepted way to scale large monolithic applications and I see no reason to adjust based on the newfangled ideas of some wannabe coders.
